### PR TITLE
Specify environment_commands

### PR DESCRIPTION
**Pull request by forsyth2**
Update v1.0.0 docs to explain that it is necessary to specify `environment_commands`. See #109 for more discussion. After that pull request merges (i.e., in future `zppy` releases), it will not be necessary to specify `environment_commands`.

### CODE REVIEW

**Comment by forsyth2** [comment]
@tomvothecoder @xylar Here are steps to get the changes in this pull request:

```
git fetch upstream
git log # Get commit for the v1.0.0 tag
git checkout -b specify-env d407b3251864c67c341a2bdf65705760c1fd1d52 # The v1.0.0 tag
```
In `docs/source/getting_started.rst`, I added `.. _activate:` above "Activate e3sm_unified environment". In `docs/source/post.mysimulation.cfg` and `docs/source/post.rrm_simulation.cfg` I replaced `e3sm_unified = latest` with `environment_commands = "source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh"`. In `docs/source/tutorial.rst`, I added "``environment_commands`` must also be updated between machines -- see :ref:`"Activate e3sm_unified environment" <activate>`." to the first paragraph.

Then:
```
cd docs
make html
mv _build ../../_build_copy
cd ../
git commit -m "Specify environment_commands" # Save changes for now; probably no point in merging these changes to `main`
git checkout -b specify-env-gh-pages upstream/gh-pages
git branch
cd _build/html
mv v1.0.0 v1.0.0_outdated # If these changes are sufficient, it probably makes sense to just delete this directory.
mv ../../../_build_copy v1.0.0
```
Are those changes sufficient to update the v1.0.0 docs (https://e3sm-project.github.io/zppy/_build/html/v1.0.0/index.html)? Does anything have to change in `docs/_build/doctrees`?

**Comment by xylar** [comment]
@forsyth2, thanks for trying this out.

It doesn't look like you're copying files into the right place.  There are a bunch of files that are showing up as moved:
```
build/html/v1.0.0/_images/git-flow.svg → _build/html/v1.0.0/html/_images/git-flow.svg
```
So there shouldn't be that extra `html` directory.  In your process, you should have done:
```
cd docs
make html
mv _build/html ../../new_v1.0.0
cd ../
# don't bother with this
# git commit -m "Specify environment_commands" # Save changes for now; probably no point in merging these changes to `main`
git checkout -b specify-env-gh-pages upstream/gh-pages
git branch
cd _build/html
mv v1.0.0 v1.0.0_outdated # If these changes are sufficient, it probably makes sense to just delete this directory.
mv ../../../new_v1.0.0 v1.0.0
```
It seems like you're also adding a bunch of files that aren't currently part of the docs.  Presumably this is because these files should be in the `.gitignore`.  These include `.buildinfo` and several python cache files like `tests/__pycache__/test_sections.cpython-37.pyc`

If everything goes right, you should see only 2 files changed, not 200+, and those files should correspond to the rst file you edited (the HTML produced from that file will change, and the text file that is the "source code" for that rst file should also change.  If the version of sphinx or your template have also changed since the v1.0.0 docs were built, you may see some css files and such get updated.  Other than that, everything should be the same.

Please give it another try and ping us if you have trouble.

**Comment by forsyth2** [comment]
After rebasing off the latest `gh-pages` and copying `_build/html` rather than `_build`, I'm down to 67 files changed. It looks like the other changes are just in formatting for html/css, but I'm not entirely sure. I do notice that `_build/html/v1.0.0/_static/underscore-1.13.1.js` becomes `_build/html/v1.0.0/_static/underscore-1.12.0.js`, which seems like a version downgrade.

**Review by forsyth2 — COMMENTED** [review]
It would also be good to update the docs to explain 1) to run each MPAS-Analysis task with the same environment and 2) how to set "environment_commands"

File: _build/html/v1.0.0/getting_started.html [hunk 1]
```diff
@@ -181,10 +179,10 @@
           <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div itemprop="articleBody">
             
-  <div class="section" id="getting-started">
+  <section id="getting-started">
 <span id="id1"></span><h1>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
-<div class="section" id="activate-e3sm-unified-environment">
-<h2>Activate e3sm_unified environment<a class="headerlink" href="#activate-e3sm-unified-environment" title="Permalink to this headline">¶</a></h2>
+<section id="activate-e3sm-unified-environment">
+<span id="activate"></span><h2>Activate e3sm_unified environment<a class="headerlink" href="#activate-e3sm-unified-environment" title="Permalink to this headline">¶</a></h2>
```
Comment (forsyth2) [inline _build/html/v1.0.0/getting_started.html:185]: Relevant change.

File: _build/html/v1.0.0/parameters.html [hunk 2]
```diff
@@ -200,6 +198,10 @@
 <span class="n">e3sm_unified</span> <span class="o">=</span> <span class="n">option</span><span class="p">(</span><span class="s1">&#39;latest&#39;</span><span class="p">,</span> <span class="s1">&#39;test&#39;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s1">&#39;latest&#39;</span><span class="p">)</span>
 <span class="c1"># This should be set to True if you don&#39;t want the batch jobs to be submitted</span>
 <span class="n">dry_run</span> <span class="o">=</span> <span class="n">boolean</span><span class="p">(</span><span class="n">default</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
+<span class="c1"># The following parameter is not defined in `default.ini`</span>
+<span class="c1"># Set up the environment -- this is where you can tell zppy to use a custom conda environment.</span>
+<span class="c1"># To use a custom conda environment, you can set `environment_commands=&quot;source &lt;path to conda.sh&gt;; conda activate &lt;custom environment&gt;&quot;`.</span>
+<span class="n">environment_commands</span> <span class="o">=</span> <span class="n">string</span>
```
Comment (forsyth2) [inline _build/html/v1.0.0/parameters.html:204]: Relevant change.

File: _build/html/v1.0.0/parameters.html [hunk 3]
```diff
@@ -357,6 +359,8 @@
...
+<span class="c1"># Note that `environment_commands` needs to be the same for all related runs of `mpas_analysis`.</span>
+<span class="c1"># For example, if years 1-50 are run using one environment and years 51-100 are run using another, MPAS-Analysis may fail.</span>
```
Comment (forsyth2) [inline _build/html/v1.0.0/parameters.html:363]: Relevant change.

File: _build/html/v1.0.0/tutorial.html [hunk 4]
```diff
@@ -168,130 +166,241 @@
...
 and <code class="docutils literal notranslate"><span class="pre">dc_obs_climo</span></code>. Different machines also have different partition names, so
-<code class="docutils literal notranslate"><span class="pre">partition</span></code> may need to be changed as well.</p>
-<div class="section" id="example-1">
+<code class="docutils literal notranslate"><span class="pre">partition</span></code> may need to be changed as well. <code class="docutils literal notranslate"><span class="pre">environment_commands</span></code> must also be
+updated between machines – see <a class="reference internal" href="getting_started.html#activate"><span class="std std-ref">“Activate e3sm_unified environment”</span></a>.</p>
```
Comment (forsyth2) [inline _build/html/v1.0.0/tutorial.html:176]: Relevant change.

File: _build/html/v1.0.0/tutorial.html [hunk 5]
```diff
@@ -168,130 +166,241 @@
...
+<span class="normal">108</span>
+<span class="normal">109</span>
+<span class="normal">110</span>
+<span class="normal">111</span></pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="k">[default]</span>
+<span class="na">input</span> <span class="o">=</span> <span class="s">&lt;input&gt;</span>
+<span class="na">input_subdir</span> <span class="o">=</span> <span class="s">archive/atm/hist</span>
+<span class="na">output</span> <span class="o">=</span> <span class="s">&lt;output&gt;</span>
+<span class="na">case</span> <span class="o">=</span> <span class="s">20210122.v2_test01.piControl.ne30pg2_EC30to60E2r2-1900_ICG.chrysalis</span>
+<span class="na">www</span> <span class="o">=</span> <span class="s">&lt;www&gt;</span>
+<span class="na">environment_commands</span> <span class="o">=</span> <span class="s">&quot;source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh&quot;</span>
```
Comment (forsyth2) [inline _build/html/v1.0.0/tutorial.html:297]: Relevant change.

File: _build/html/v1.0.0/tutorial.html [hunk 6]
```diff
@@ -306,95 +415,175 @@ <h2>Example 1<a class="headerlink" href="#example-1" title="Permalink to this he
...
+<span class="normal">78</span>
+<span class="normal">79</span>
+<span class="normal">80</span>
+<span class="normal">81</span></pre></div></td><td class="code"><div class="highlight"><pre><span></span><span class="k">[default]</span>
+<span class="na">input</span> <span class="o">=</span> <span class="s">&lt;input&gt;</span>
+<span class="na">input_subdir</span> <span class="o">=</span> <span class="s">archive/atm/hist</span>
+<span class="na">output</span> <span class="o">=</span> <span class="s">&lt;output&gt;</span>
+<span class="na">case</span> <span class="o">=</span> <span class="s">20210603.v2rc3c.piControl.northamericax4v1pg2_WC14to60E2r3.chrysalis</span>
+<span class="na">www</span> <span class="o">=</span> <span class="s">&lt;www&gt;</span>
+<span class="na">environment_commands</span> <span class="o">=</span> <span class="s">&quot;source /lcrc/soft/climate/e3sm-unified/load_latest_e3sm_unified_chrysalis.sh&quot;</span>
```
Comment (forsyth2) [inline _build/html/v1.0.0/tutorial.html:508]: Relevant change.

**Review by forsyth2 — COMMENTED** [review]
https://github.com/forsyth2/zppy/pull/2 shows the relevant changes in a more concise form. These are the related HTML changes.

**Comment by forsyth2** [comment]
Closing in favor of #114.

**Pull request closed without merging**